samples/batch-synthesis/csharp/BatchSynthesisSample/BatchSynthesisClient.cs

```diff
@@ -0,0 +1,115 @@
+//
+
+using Newtonsoft.Json;
+using System.Net.Http.Formatting;
+
+public class BatchSynthesisClient
+{
+    private const string OcpApimSubscriptionKey = "Ocp-Apim-Subscription-Key";
+
+    private readonly string hostName;
+    private readonly string baseUri;
+    private readonly string subscriptionKey;
+
+    private readonly HttpClient client;
+
+    public BatchSynthesisClient(string hostName, string key)
+    {
+        this.hostName = hostName;
+        this.subscriptionKey = key;
+        this.baseUri = $"{this.hostName}/api/texttospeech/3.1-preview1/batchsynthesis";
+
+        this.client = new HttpClient();
+        client.DefaultRequestHeaders.Add(OcpApimSubscriptionKey, this.subscriptionKey);
+    }
+
+    public async Task<IEnumerable<BatchSynthesis>> GetAllSynthesesAsync()
+    {
+        var syntheses = new List<BatchSynthesis>();
+        var uri = new Uri(this.baseUri);
+        do
+        {
+            var response = await this.client.GetAsync(uri).ConfigureAwait(false);
+
+            if (!response.IsSuccessStatusCode)
+            {
+                await HandleErrorResponse(response);
+                return syntheses;
+            }
+
+            var pagedSyntheses = await response.Content.ReadAsAsync<PaginatedResults<BatchSynthesis>>().ConfigureAwait(false);
+            syntheses.AddRange(pagedSyntheses.Values);
+            uri = pagedSyntheses.NextLink;
+        }
+        while (uri != null);
+
+        return syntheses;
+    }
+
+    public async Task<BatchSynthesis> GetSynthesisAsync(Guid id)
+    {
+        var uri = new Uri(this.baseUri + $"/{id}");
+        var response = await this.client.GetAsync(uri).ConfigureAwait(false);
+        if (!response.IsSuccessStatusCode)
+        {
+            await HandleErrorResponse(response);
+            return null;
+        }
+
+        return await response.Content.ReadAsAsync<BatchSynthesis>().ConfigureAwait(false);
+    }
+
+    public async Task DeleteSynthesisAsync(Guid id)
+    {
+        var uri = new Uri(this.baseUri + $"/{id}");
+        var response = await this.client.DeleteAsync(uri).ConfigureAwait(false);
+        if (!response.IsSuccessStatusCode)
+        {
+            await HandleErrorResponse(response);
+        }
+    }
+
+    public async Task<Uri> CreateSynthesisAsync(
+        string voiceName,
+        string displayName,
+        string description,
+        string script,
+        bool isSsml)
+    {
+        var uri = new Uri(this.baseUri);
+
+        var batchSynthesis = new BatchSynthesis
+        {
+            DisplayName = displayName,
+            Description = description,
+            TextType = isSsml ? "Ssml" : "PlainText",
+            SynthesisConfig = new BatchSynthesisConfig
+            {
+                Voice = voiceName
+            },
+            Inputs = new List<BatchSynthesisInputDefinition> { new BatchSynthesisInputDefinition { Text = script } }
```

Comment:
For documentation at learn.microsoft.com, does the max length of "Inputs" == 1,000? Each text input can have several paragraphs, but those paragraphs don't count against the 1,000 limit?

Reply:
Each job could has multi inputs at the maximum of 1000, i.e. the list length in this line <= 1000. We don't care the paragraph (i.e. line break) in each input script. And, the whole payload should < 500KB.